packages/core/src/new-api/internal/execution/onchain-state-transitions.ts

```diff
@@ -619,3 +587,43 @@ async function _readEventArg(
     };
   }
 }
+
+async function checkTransactionComplete(
+  state: ExecutionEngineState,
+  next: { txHash: string; futureId: string; executionId: number },
+  successConstructor: (
+    receipt: ethers.providers.TransactionReceipt
+  ) => ExecutionSuccess | ExecutionFailure | TransactionMessage
+): Promise<OnchainStateTransitionContinue | OnchainStateTransitionPause> {
+  const currentTransaction = await state.chainDispatcher.getTransaction(
+    next.txHash
+  );
+
+  if (currentTransaction === null || currentTransaction === undefined) {
+    throw new IgnitionError(
+      `Transaction ${next.txHash} (${next.futureId}/${next.executionId}) has dropped from mempool`
+    );
+  }
+
+  const receipt = await state.chainDispatcher.getTransactionReceipt(
+    next.txHash
+  );
+
+  if (receipt === null || receipt === undefined) {
+    // No receipt means it hasn't been included in a block yet
+    return { status: "pause" };
+  }
```

Comment:
This information is already in the `currentTransaction` object, so the `getTransactionReceipt` call can be saved.

Reply:
Is it possible to get the contractAddress from a deploy transaction object? It is only needed for deploys rather than calls/sends, so we could split out two different versions even if we can't remove it for deploy.